sources/packages/backend/apps/api/src/services/education-program-offering/education-program-offering-validation.models.ts

```diff
@@ -183,23 +183,73 @@ export type CalculatedStudyBreaksAndWeeks = StudyBreaksAndWeeks & {
  */
 export enum OfferingValidationWarnings {
   InvalidStudyBreakAmountOfDays = "invalidStudyBreakAmountOfDays",
-  InvalidStudyBreaksCombinedThresholdPercentage = "invalidStudyBreaksCombinedThresholdPercentage",
   ProgramOfferingIntensityMismatch = "programOfferingIntensityMismatch",
   ProgramOfferingDeliveryMismatch = "programOfferingDeliveryMismatch",
   ProgramOfferingWILMismatch = "programOfferingWILMismatch",
   InvalidStudyDatesPeriodLength = "invalidStudyDatesPeriodLength",
 }
 
 /**
- * Represent the context of an error that should be considered a warning.
+ * Possible information unique identifiers.
+ * !These keys are also consumed in the UI to display/hide info banners.
+ */
+export enum OfferingValidationInfos {
+  InvalidStudyBreaksCombinedThresholdPercentage = "invalidStudyBreaksCombinedThresholdPercentage",
+}
+
+/**
+ * Represent the context of an error that has an additional context to
+ * express a possible warning or some relevant information captured during
+ * the validation process.
  * All validations when failed will generate an error. The ones that have
- * the warning context will be considered as warnings then.
+ * the warning or information contexts will not be considered critical.
  */
-export class ValidationWarning {
-  constructor(public readonly warningType: OfferingValidationWarnings) {
-    this.isWarning = true;
+export class ValidationContext {
+  /**
+   * Creates an error context that will make the error downgrade to
+   * a condition of a warning information will force the offering
+   * to be reviewed by the Ministry.
+   * @param warningTypeCode warning code that uniquely identifies this condition.
+   * @returns the warning context.
+   */
+  static CreateWarning(
```

Comment:
Wouldn't a constructor here be bit more easier than 2 static methods here?

Reply:
We would need a constructor accepting the two types of enums and/or a constructor and 2 overloads, which can be achieved in typescript but is a bit messy. For this one, it seems much more straightforward to implement (and consume it) in the way it is.